francis/output_progress.py

```diff
@@ -35,3 +36,7 @@ def __exit__(self, exception, value, tb):
         time.sleep(self.delay)
         if exception is not None:
             return False
+
+
+def default_bar(text, bar_length):
```

Comment:
I know this is different from what you said about passing a bar object as an argument, however I could not think of a way to make that work as each function that uses the bar, has different parameters such as its name and its length. Therefore instead, I have created a function which returns an object, and by changing some of its defaults i.e. the fill will impact all the bars.

Reply:
What you have done is almost what I had in mind. 
I was thinking something along the lines of ...
```
processing_bar = default_bar("processing", len(pre_df.index))
pre_df = processing.process(pre_df,
                CONFIG["SAMPLE_RATE"],
                CONFIG["PREPROCESS"],
                bar_config=processing_bar)
```
This separates the logic from the view as much as possible.
But what you have done is an improvement to how it was, so its in the right direction.

One thing I'm not too big a fan of is passing in `default_bar` as the default argument, that means any situation that you **must always define a default_bar** or import when you might not want one in the first place. 

You should explicitly pass in the default bar,

 if you dont give a bar config it should default to `None`, something like this. And then have a way of handling a None type for bar_config.

Something like this ...
```
def process(df, sample_rate, pre_process=True, bar_config=None):
```